CRM/SMS/BAO/Provider.php

```diff
@@ -88,30 +88,41 @@ public static function getProviders($selectArr = NULL, $filter = NULL, $getActiv
   }
 
   /**
-   * Save a new record into the database
-   * @todo create a create function to do this work
-   * @param $values
+   * Create or Update an SMS provider
+   * @param array $params
+   * @param array $ids to update
+   * @return array saved values
    */
-  public static function saveRecord($values) {
-    $values['domain_id'] = CRM_Utils_Array::value('domain_id', $values, CRM_Core_Config::domainID());
-    $dao = new CRM_SMS_DAO_Provider();
-    $dao->copyValues($values);
-    $dao->save();
-  }
+  public static function create($params, $ids = array()) {
+    $id = CRM_Utils_Array::value('id', $ids, CRM_Utils_Array::value('id', $params));
```

Comment:
seems like as per https://github.com/civicrm/civicrm-core/blob/master/api/v3/utils.php#L1281 it expects a 2nd param which is ids